### PR TITLE
feat: update license at settings page

### LINKED ISSUE #1099: Show license setting in settings page
Allow member to update the license from the settings page.

We keep the license setting in the library as well. So it would be best to reuse the same component in both pages.

### PR DESCRIPTION
**Pull request by LinaYahya**
closes #1099 

image

image
image

### CODE REVIEW

File: src/components/item/publish/CCLicenseDialog.tsx [hunk 1]
```diff
@@ -15,55 +15,58 @@ import CancelButton from '../../common/CancelButton';
...
+  disableSubmission?: boolean;
 };
 
 const CCLicenseDialog = ({
   open,
   setOpen,
-  disabled = false,
-  buttonName,
   handleSubmit,
+  isNewDialog = true,
```
Comment (pyphilia) [inline src/components/item/publish/CCLicenseDialog.tsx:28]: I think it's cleaner if you create a new component `CCLicenseModal` that uses `CCLicenseDialog`. I would avoid having `isNewDialog` everywhere.

File: src/components/item/publish/CCLicenseSelection.tsx [hunk 2]
```diff
@@ -124,132 +137,160 @@ const CCLicenseSelection = ({ item, disabled }: Props): JSX.Element => {
...
+
+      {confirmSubmitInNewDialog && (
+        <Button
+          variant="contained"
+          onClick={() => setOpen(true)}
+          disabled={disabled || !requireAttributionValue} // disable the button if no option is selected
+        >
+          {translateBuilder(BUILDER.ITEM_SETTINGS_CC_LICENSE_SUBMIT_BUTTON)}
+        </Button>
+      )}
```
Comment (pyphilia) [inline src/components/item/publish/CCLicenseSelection.tsx:275]: Could it go to the `CCLicenseModal` new component?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR! I think it needs some refactoring to differentiate layout with components rather than props. Let me know if you have questions!

File: src/components/item/publish/useItemLicense.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    useState<CCLicenseChoice>('');
+  const [allowCommercialValue, setAllowCommercialValue] =
+    useState<CCLicenseChoice>('');
+  const [allowSharingValue, setAllowSharingValue] =
+    useState<CCSharingLicenseChoice>('');
+  // itemId
+  const itemId = item?.id;
+
+  const settings = item?.settings;
+  const itemName = item?.name;
```
Comment (spaenleh) [inline src/components/item/publish/useItemLicense.tsx:44]: Use destructuring

File: src/components/item/publish/useItemLicense.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      setAllowCommercialValue(allowCommercialUse ? 'yes' : 'no');
+      setRequireAttributionValue(requireAccreditation ? 'yes' : 'no');
+    }
+  }, [settings]);
+
+  const handleSubmit = () => {
+    if (requireAttributionValue) {
+      updateCCLicense({
+        id: itemId,
+        name: itemName,
```
Comment (spaenleh) [inline src/components/item/publish/useItemLicense.tsx:62]: Do you have to specify the name when you update the item settings ??

**Review by spaenleh — APPROVED** [review]
Looks good, the refactoring looks nice !

File: src/components/item/publish/useItemLicense.tsx (hunk 4 above)
Comment (LinaYahya) [reply src/components/item/publish/useItemLicense.tsx:62]: No need at all

**Review by spaenleh — APPROVED** [review]
🚀

**Review by pyphilia — APPROVED** [review]

**Pull request merged**